UserRepository/Tests/UserRepositoryTests.swift

```diff
@@ -0,0 +1,188 @@
+    }
+
+    override func tearDown() {
+        UserRepositoryFactory.clearInstance()
+        if Hippolyte.shared.isStarted {
+            Hippolyte.shared.stop()
+        }
+    }
+
+    func testGetUser_individualSuccess() {
+        // Given
+        let expectation = self.expectation(description: "Get HyperwalletUser completed")
+        HyperwalletTestHelper.setUpMockServer(request: UserRequestHelper.setUpRequest(individualUserResponse))
+
+        var user: HyperwalletUser?
+
+        // When
+        userRepository.getUser { result in
+            switch result {
+            case .success(let userResponse):
+                guard let userResponse = userResponse else {
+                    XCTFail("The userResponse should not be empty!")
+                    return
+                }
+
+                expectation.fulfill()
+                user = userResponse
+
+            case .failure:
+                XCTFail("Unexpected error")
+            }
+        }
+        wait(for: [expectation], timeout: 1)
+        // Then
+        verifyIndividualResponse(user)
+    }
+
+    func testGetUser_businessSuccess() {
+        // Given
+        let expectation = self.expectation(description: "Get HyperwalletUser completed")
+        HyperwalletTestHelper.setUpMockServer(request: UserRequestHelper.setUpRequest(businessUserResponse))
+
+        var user: HyperwalletUser?
+
+        // When
+        userRepository.getUser { result in
+            switch result {
+            case .success(let userResponse):
+                guard let userResponse = userResponse else {
+                    XCTFail("The userResponse should not be empty!")
+                    return
+                }
+
+                expectation.fulfill()
+                user = userResponse
+
+            case .failure:
+                XCTFail("Unexpected error")
+            }
+        }
+        wait(for: [expectation], timeout: 1)
+        // Then
+        verifyBusinessResponse(user)
+    }
+
+    func testGetUser_failure() {
+        // Given
+        let expectation = self.expectation(description: "Get HyperwalletUser completed")
+        let error = NSError(domain: NSURLErrorDomain, code: 501, userInfo: nil)
+        HyperwalletTestHelper.setUpMockServer(request: UserRequestHelper.setUpRequest(individualUserResponse,
+                                                                                      error))
+        // When
+        userRepository.getUser { result in
+            switch result {
+            case .success:
+                XCTFail("The request should return error!")
+
+            case .failure:
+                expectation.fulfill()
+            }
+        }
+        wait(for: [expectation], timeout: 1)
+    }
+}
+
+private extension UserRepositoryTests {
+    func verifyIndividualResponse(_ user: HyperwalletUser?) {
+        XCTAssertNotNil(user)
+
+        XCTAssertEqual(user!.clientUserId, "myAppUserId01")
```

Comment:
add generic text message to all the asserts

Reply:
Messages added